File: DSA/Raima_Rekhi_590013023/Day11Question3.c
#include <stdio.h>
#include <string.h>

int count_palindromic_substrings(char str[]);

int main(){
char str[1000];

//Prompt the user to enter a string
printf("Enter a string:\n");
fgets(str,sizeof(str),stdin);
str[strcspn(str,"\n")]='\0';

int total=count_palindromic_substrings(str);
printf("Total number of palindromic substrings: %d\n",total);
return 0;
}

// Function to count all palindromic substrings
int count_palindromic_substrings(char str[]){
int n=strlen(str);
int count=0;

// Loop through each possible center of a palindrome
for(int center=0;center<2*n-1;center++){
int left=center/2;
int right=left+(center%2);

// Expand outward as long as characters match (valid palindrome)
while(left>=0 && right<n && str[left]==str[right]){
count++;
left--;
right++;
}
}
return count;
}

File: DSA/VaniAgrawal_590015846/Day_11_Question_3.py
#Count the total number of palindromic substrings in a string
s = input("enter string :")

n = 0
for _ in s:
    n += 1

count = 0
i = 0

while i < n:
    left = i
    right = i
    while left >= 0 and right < n and s[left] == s[right]:
        count += 1
        left -= 1
        right += 1

    left = i
    right = i + 1
    while left >=0 and right < n and s[left]==s[right]:
        count += 1
        left -= 1
        right +=1

    i+=1

print(count)
